pkg/splunk/enterprise/afwscheduler.go

```diff
@@ -869,6 +877,131 @@ func afwGetReleventStatefulsetByKind(cr splcommon.MetaObject, client splcommon.C
 	return sts
 }
 
+// getIdxcPodExecClient returns the client object used to execute pod exec commands for idxc
+func getIdxcPodExecClient(client splcommon.ControllerClient, cr splcommon.MetaObject, applyIdxcBundleCmdStr string, targetPodName string) *IdxcPodExecClient {
+	scopedLog := log.WithName("getIdxcPodExecClient").WithValues("crName", cr.GetName(), "namespace", cr.GetNamespace())
+
+	// get the admin password from the namespace scoped secret
+	adminPwd, err := getAdminPasswrdFromSecret(client, cr)
+	if err != nil {
+		scopedLog.Error(err, "unable to get admin password")
+		return nil
+	}
+
+	return &IdxcPodExecClient{
+		client:        client,
+		cr:            cr,
+		targetPodName: targetPodName,
+		adminPwd:      adminPwd,
+		command:       fmt.Sprintf(applyIdxcBundleCmdStr, adminPwd),
+	}
+}
+
+// getIdxcPlayBookContext returns the idxc playbook context
+func getIdxcPlayBookContext(client splcommon.ControllerClient, cr splcommon.MetaObject, afwPipeline *AppInstallPipeline, podName string) *IdxcPlayBookContext {
+	return &IdxcPlayBookContext{
+		client:        client,
+		cr:            cr,
+		afwPipeline:   afwPipeline,
+		targetPodName: podName,
+		podExecClient: getIdxcPodExecClient(client, cr, applyIdxcBundleCmdStr, podName),
```

Comment:
nit: do we need to check nil value for podExecClient, to avoid invalid mem references later?